src/main/java/week5/AlphabetFinderTest.java

```diff
@@ -0,0 +1,102 @@
+package week5;
+
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.Test;
+
+public class AlphabetFinderTest {
+
+  @Test
+  public void testEmpty() {
+    assertThat(AlphabetFinder.findAlphabet(null), is(Collections.EMPTY_LIST));
+    assertThat(AlphabetFinder.findAlphabet(new String[]{}), is(Collections.EMPTY_LIST));
+  }
+
+  @Test
+  public void testExample() {
+    //test example that was given
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "CAT", "CAR"}),
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
+
+  @Test
+  public void testContradiction() {
+    //does the program crash on contradiction
+    assertThat(AlphabetFinder.findAlphabet(new String[]{"ART", "RAT", "ARC", "CAT", "CAR"}),
+        is(Arrays.asList('A', 'T', 'R', 'C')));
+  }
```

Comment:
The problem here, is that I was removing duplicates when inferring the partial order so in this case [A, R, A, C] -> [A, R, C]. So updated the test and implementation so in the case of cycle the algorithm returns an Optional.empty